### PR TITLE
Add maven reporting

### PR DESCRIPTION
**Pull request by kevinhagel**
Added vavr.io library for better exception handling, better try-with-resources handling.
Refactored methods
cleaned up many of the commit warnings and errors

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+    "java.configuration.updateBuildConfiguration": "automatic",
+    "java.compile.nullAnalysis.mode": "disabled",
+    "java.format.settings.url": "https://raw.githubusercontent.com/google/styleguide/gh-pages/eclipse-java-google-style.xml",
+    "java.format.settings.profile": "GoogleStyle",
+    "java.checkstyle.configuration": "/google_checks.xml",
+    "org.eclipse.jdt.core.formatter.join_wrapped_lines" : false,
+    "cSpell.words": [
```
Comment (hyyan) [inline .vscode/settings.json:8]: Probably , you copied this form the engine , but I don;t this you need this

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      "radiobutton"
+    ],
+    "[markdown]": {
```
Comment (hyyan) [inline .vscode/settings.json:11]: Same for the markdown section

File: src/test/java/org/dwcj/maven/install/DwcjInstallIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
+class DwcjInstallIntegrationTest {
```
Comment (hyyan) [inline src/test/java/org/dwcj/maven/install/DwcjInstallIntegrationTest.java:15]: What is the purpose of this ?

File: src/test/java/org/dwcj/maven/install/DwcjInstallTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import static org.mockito.Mockito.when;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * This is a test class
+ */
+@ExtendWith(MockitoExtension.class)
+class DwcjInstallTest {
```
Comment (hyyan) [inline src/test/java/org/dwcj/maven/install/DwcjInstallTest.java:18]: It does not seem this is testing anything

**Review by hyyan — CHANGES_REQUESTED** [review]
@kevinhagel It appears that this pull request (PR) is addressing multiple issues simultaneously, and the title of the PR does not accurately reflect its content. I've noticed that you have made modifications to the POM file and also removed MimeMultipartData.java. Could you please close this PR and submit the changes in smaller, incremental patches?

**Comment by kevinhagel** [comment]
I will start this over from a branch off of trunk.  This was experimental anyway, as far as the maven reporting is concerned.  i was generating an example of how the site plugin works, how we can generate the kind of site any maven plugin has.
I will remove that idea for another time.
The other primary update is to move to a more modern version of http components.  This method explains why were aren't using MimeMultiPartData.java anymore
 ```
 /**
   * The {@link MultipartEntityBuilder} creates a builder whose build method is closeable.
   *
   * @param file the file being sent.
   * @return an HttpEntity created by using the builder
   * @throws IOException if we can't probe the content type of the file.
   */
  protected HttpEntity createMultipartEntity(File file) throws IOException {
    ContentType mimetype = ContentType.create(Files.probeContentType(file.toPath()));
    getLog().info("Installing file " + file + ", mimetype = " + mimetype);
    FileBody fileBody = new FileBody(file, mimetype);

    return MultipartEntityBuilder.create()
        .setMode(HttpMultipartMode.EXTENDED)
        .addPart("jar", fileBody)
        .setCharset(StandardCharsets.UTF_8)
        .setContentType(ContentType.MULTIPART_FORM_DATA)
        .build();
  }
```
much smaller, much more robust, uses modern http core components.
I'm also adding the use of the vavr library for a more functional programming model for Try, Try.withRessources.

**Comment by hyyan** [comment]
@kevinhagel I fully support that idea, but let's keep enhancements in separate pull requests, focusing on only one feature at a time.

**Comment by kevinhagel** [comment]
For all of these issues I have created a new feature branch, concentrating on the http components update and the required update to DwcjInstall and the pom.  I'm getting ready to commit that and create a PR

File: src/test/java/org/dwcj/maven/install/DwcjInstallTest.java (hunk 4 above)
Comment (kevinhagel) [reply src/test/java/org/dwcj/maven/install/DwcjInstallTest.java:18]: removed

File: .vscode/settings.json (hunk 2 above)
Comment (kevinhagel) [reply .vscode/settings.json:11]: removed .vscode

**Comment by kevinhagel** [comment]
I would like to delete this branch, I have the code on my machine if we want to get involved with maven reporting again.  The original idea was just to do that.  I've seen no interest in it here at this point, so let's move that off into some future when we have the time.

**Comment by hyyan** [comment]
Sounds good. closed

**Pull request closed without merging**